Python_Scripts/script_regroup_models.py

```diff
@@ -0,0 +1,337 @@
+
+    return model_name, model_year
+
+
+def _check_biomodels_model(sedml_model, sbml_path, model_name, model_year, model_info):
+    # only one sbml file, a benchmark model on its own.
+    sedml_path = ''
+    sbml_model = (libsbml.readSBML(sbml_path)).getModel()
+    n_species = len(sbml_model.getListOfSpecies())
+    n_reactions = len(sbml_model.getListOfReactions())
+
+    # get the simulation times and write them to csv file
+    out_start, out_end, n_timepoints, _ = \
+        timePointsBioModels(sedml_model)
+    if out_start is None:
+        return model_info
+
+    model_info.append({
+        'name': model_name,
+        'year': model_year,
+        'n_species': n_species,
+        'n_reactions': n_reactions,
+        'long_id': (model_name, model_year, n_species, n_reactions),
+        'short_id': '',
+        'sbml_path': sbml_path,
+        'sedml_path': sedml_path,
+        'regrouped_path': '',
+        'start_time': out_start,
+        'end_time': out_end,
+        'n_timepoints': n_timepoints,
+        'sedml_task': '',
+    })
+
+    return model_info
+
+
+def _check_sedml_submodels(sedml_file, sbml_files,
+                           model_name, model_year, model_info):
+    """Checks which sbml models of one sedml group belong together"""
+    sedml_path = os.path.abspath(sedml_file)
+    sedml_doc = libsedml.readSedML(sedml_file)
+    sedml_task_models = [task.getModelReference() for task in sedml_doc.getListOfTasks()]
+
+    for sbml_file in sbml_files:
+        # we want to find a simulation task for this SBML model
+        # first, get the name of the file (identifier in SED-ML)
+        sbml_path = os.path.abspath(sbml_file)
+        sbml_file_short = (sbml_file.split('/')[-1]).split('.')[0]
+
+        # Find the first task for this model (and warn if no task was found)
+        try:
+            task_number = sedml_task_models.index(sbml_file_short)
+        except ValueError:
+            print('SBML file not found in list of tasks: ' + sbml_file_short)
+            return model_info
+
+        # get the corresponding simulation setting
+        current_sim = sedml_doc.getSimulation(sedml_doc.getTask(
+            task_number).getSimulationReference())
+        # remember simulation settings
+        out_start = current_sim.getOutputStartTime()
+        out_end = current_sim.getOutputEndTime()
+        n_timepoints = current_sim.getNumberOfPoints()
+
+        # get info about the SBML file
+        sbml_model = libsbml.readSBML(sbml_file).getModel()
+        if sbml_model is None:
+            # check for SBML model
+            print('No sbml model found for file' + sbml_file +
+                  ', in model name ' + model_name +
+                  ', model year ' + model_year)
+            return model_info
+        n_species = len(sbml_model.getListOfSpecies())
+        n_reactions = len(sbml_model.getListOfReactions())
+
+        # add model information to the list
+        model_info.append({
```

Comment:
refactor: here you pass and return model_info. but internally you don't need it, right? then just create that dict and put the append outside the function.

Reply:
makes it clearer.

Reply:
ah sorry, I forgot the for loop (it's getting late ... :sleeping: ) yeah, fine as it is.